web-services/query/src/test/java/datawave/webservice/query/runner/ExtendedQueryExecutorBeanTest.java

```diff
@@ -864,6 +865,161 @@ public void testCreateQueryAndNext_HappyPath() throws Exception {
+        this.queryLogic1.setupQuery(this.genericConfiguration);
+        expect(this.queryLogic1.getTransformIterator(this.query)).andReturn(this.transformIterator);
+        cache.put(eq(queryId.toString()), isA(RunningQuery.class));
+        expect(this.genericConfiguration.getQueryString()).andReturn(queryName).once();
+        expect(this.qlCache.poll(queryId.toString())).andReturn(null);
+        
+        // Set expectations of the next logic
+        expect(this.principal.getName()).andReturn(userName);
+        expect(this.principal.getShortName()).andReturn(userSid);
+        expect(this.context.getUserTransaction()).andReturn(this.transaction).anyTimes();
+        expect(this.transaction.getStatus()).andReturn(Status.STATUS_NO_TRANSACTION).anyTimes();
+        cache.unlock("{id}");
+        PowerMock.expectLastCall().anyTimes();
+        this.transaction.begin();
+        this.transaction.setRollbackOnly();
+        PowerMock.expectLastCall().anyTimes();
+        
+        expect(this.responseObjectFactory.getEventQueryResponse()).andReturn(new DefaultEventQueryResponse());
+        this.transaction.commit();
+        QueryExecutorBean subject = new QueryExecutorBean();
+        // Run the test
+        PowerMock.replayAll();
+        
+        setInternalState(subject, EJBContext.class, context);
+        setInternalState(subject, AccumuloConnectionFactory.class, connectionFactory);
+        setInternalState(subject, ResponseObjectFactory.class, responseObjectFactory);
+        setInternalState(subject, CreatedQueryLogicCacheBean.class, qlCache);
+        setInternalState(subject, QueryCache.class, cache);
+        setInternalState(subject, ClosedQueryCache.class, closedCache);
+        setInternalState(subject, Persister.class, persister);
+        setInternalState(subject, QueryLogicFactoryImpl.class, queryLogicFactory);
+        setInternalState(subject, QueryExpirationConfiguration.class, queryExpirationConf);
+        setInternalState(subject, AuditBean.class, auditor);
+        setInternalState(subject, QueryMetricsBean.class, metrics);
+        setInternalState(subject, Multimap.class, traceInfos);
+        setInternalState(subject, SecurityMarking.class, new ColumnVisibilitySecurityMarking());
+        setInternalState(subject, QueryParameters.class, new QueryParametersImpl());
+        setInternalState(subject, QueryMetricFactory.class, new QueryMetricFactoryImpl());
+        setInternalState(connectionRequestBean, EJBContext.class, context);
+        setInternalState(subject, AccumuloConnectionRequestBean.class, connectionRequestBean);
+        subject.createQuery(queryLogicName, queryParameters);
+        
+        Throwable result1 = null;
+        try {
+            subject.next("{id}");
+        } catch (Exception e) {
+            result1 = e.getCause();
+        }
+        
+        assertNotNull("Expected a non-null response", result1);
```

Comment:
Maybe it's worth verifying that the exception is of type BadRequestQueryException with a getErrorCode equal to DatawaveErrorCode.INVALID_QUERY_ID

Reply:
Okay, will adjust this on next commit